borg/archiver.py

```diff
@@ -503,9 +503,23 @@ def do_rename(self, args, repository, manifest, key, cache, archive):
     def do_delete(self, args, repository):
         """Delete an existing repository or archive"""
         if args.location.archive:
+            archive_name = args.location.archive
             manifest, key = Manifest.load(repository)
+
+            if args.forced == 2:
+                try:
+                    del manifest.archives[archive_name]
+                except KeyError:
+                    raise Archive.DoesNotExist(archive_name)
+                logger.info('Archive deleted.')
+                manifest.write()
+                # note: might crash in compact() after committing the repo
+                repository.commit()
+                logger.info('Done. Run "borg check --repair" to clean up the mess.')
+                return self.exit_code
```

Comment:
I changed the error reporting here to match the the singular-behaviour that `borg delete` normally has in 1.0.x (RC=2 if archive not found, no plural messages etc.)